### PR TITLE
Adds more mocked tests

### PR DESCRIPTION
**Pull request by cr0ax**
A follow-up to https://github.com/containers/youki/pull/706

<a href="https://gitpod.io/#https://github.com/containers/youki/pull/711"></a>

### CODE REVIEW

**Comment by cr0ax** [comment]
I think this is the better PR, I will try to annotate it in the near future

File: crates/libcgroups/src/v2/devices/mocks.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    pub fn bpf_prog_attach(
+        _prog_fd: ::std::os::raw::c_int,
+        _attachable_fd: ::std::os::raw::c_int,
+        _type_: libbpf_sys::bpf_attach_type,
+        _flags: ::std::os::raw::c_uint,
+    ) -> ::std::os::raw::c_int {
+        unimplemented!();
+    }
+}
```
Comment (cr0ax) [inline crates/libcgroups/src/v2/devices/mocks.rs:60]: I needed a stubbed module to apply the mock to.

Do we already do something like this elsewhere?

File: crates/libcgroups/src/v2/devices/mod.rs [hunk 2]
```diff
@@ -3,4 +3,9 @@ pub mod controller;
 pub mod emulator;
 pub mod program;
 
+#[cfg(test)]
+#[allow(clippy::too_many_arguments)]
+#[cfg_attr(coverage, feature(no_coverage))]
+pub mod mocks;
```
Comment (cr0ax) [inline crates/libcgroups/src/v2/devices/mod.rs:9]: only used in tests, also libc functions trigger the `too_many_arguments` clippy warning

File: crates/libcgroups/src/v2/devices/bpf.rs [hunk 3]
```diff
@@ -11,13 +11,33 @@ pub mod prog {
...
+    };
+    #[cfg(not(test))]
+    use libbpf_sys::{
+        bpf_load_program, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
+    };
 
+    pub fn load(license: &str, insns: &[u8]) -> Result<RawFd> {
+        let insns_cnt = insns.len() / std::mem::size_of::<bpf_insn>();
+        let insns = insns as *const _ as *const bpf_insn;
+        #[allow(unused_unsafe)]
```
Comment (cr0ax) [inline crates/libcgroups/src/v2/devices/bpf.rs:37]: I don't like this one, but the mocked stuff isn't actually unsafe.

We could probably make it look `unsafe` though...

File: crates/libcgroups/src/v2/devices/bpf.rs [hunk 4]
```diff
@@ -11,13 +11,33 @@ pub mod prog {
...
+
+    // TODO: consider use of #[mockall_double]
+    #[cfg(test)]
+    use crate::v2::devices::mocks::mock_libbpf_sys::{
+        bpf_load_program, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
+    };
+    #[cfg(not(test))]
+    use libbpf_sys::{
+        bpf_load_program, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
+    };
```
Comment (cr0ax) [inline crates/libcgroups/src/v2/devices/bpf.rs:32]: For the above, we are importing mocked versions or not based on if we are in a test environment

**Review by cr0ax — COMMENTED** [review]
Added some annotations for things I want to call out for discussion.

**Review by utam0k — APPROVED** [review]
LGTM

**Pull request merged**